packages/components/src/icon/index.js

```diff
@@ -34,6 +34,7 @@ function Icon( { icon = null, size, className } ) {
 			width: iconSize,
 			height: iconSize,
 			...icon.props,
+			...style,
```

Comment:
Is this meant to be `...` spread into the _top-level_ props?

Reply:
This is on purpose. The idea here is to be able to overwrite any properties using the following construct:
```jsx
<Icon icon={ svgIcon } style={ newSvgStyles } />;
```
That includes `width` and `height` that we'd like to set to `"100%"` for instance so they fit our parent container. Or the `fill` property that could already be present on the original svg and that we might want to overwrite.

Same for `Dashicon`, we need to spead `style` so they are transformed into `extraProps` here: https://github.com/WordPress/gutenberg/blob/master/packages/components/src/dashicon/index.js#L907

Reply:
> This is on purpose. The idea here is to be able to overwrite any `SVG` attributes using the following construct:

My initial reaction here is that we're conflating style properties with DOM attributes, and that there isn't a guaranteed overlap, just like we can't have `<div background-image="..." />`.

But then I found this interesting mention in the specification, which seems specific to SVG:

>For each styling property defined in this specification (see Property Index), there is a corresponding XML attribute (the presentation attribute) with the same name that is available on all relevant SVG elements.

https://www.w3.org/TR/SVG11/styling.html#UsingPresentationAttributes

So I guess it _is_ fair to assume any style property can be assigned as a DOM attribute.

It still seems misleading to accept this as a named `style` object, only then to proceed to apply it as spread on the rendered element, rather than to just pass through all additional props from the `Icon` component (e.g. `...additionalProps` pattern) to the rendered SVG.

That being said, since I was apparently under-informed about how style properties map to DOM attributes for SVG, maybe this is an expected behavior.

Reply:
@aduth I think I got your point on this, I agree, having a prop just to use for spreading at the same time naming it as `style` doesn't make that much sense on this level. I have changed the code, would you mind giving it another look?

I think we got to that point because we are passing objects from our `.native.scss` files as a value to it. But actually the `style` here doesn't correspond to the `style` in react-native, so it doesn't need to have that name.